<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Filter_Function</class>
 <widget class="QWidget" name="Filter_Function">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1122</width>
    <height>802</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Filter_Function</string>
  </property>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>341</width>
     <height>81</height>
    </rect>
   </property>
   <property name="title">
    <string>数据链接</string>
   </property>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>30</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>输入图像：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>30</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>采集图像</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_2">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>230</y>
     <width>341</width>
     <height>231</height>
    </rect>
   </property>
   <property name="title">
    <string>参数设置</string>
   </property>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>30</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>方框滤波参数：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>70</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>均值滤波参数：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>110</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>高斯滤波参数：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>150</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>中值滤波参数：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>190</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>双边滤波参数：</string>
    </property>
   </widget>
   <widget class="QSlider" name="BoxFilterlSlider">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>30</y>
      <width>211</width>
      <height>19</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="MeanBlurSlider">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>70</y>
      <width>211</width>
      <height>19</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="GaussianBlurSlider">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>110</y>
      <width>211</width>
      <height>19</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="MedianBlurSlider">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>150</y>
      <width>211</width>
      <height>19</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="BilateralFilterSlider">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>190</y>
      <width>211</width>
      <height>19</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>10</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>50</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>90</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>130</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>170</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_3">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>130</y>
     <width>341</width>
     <height>81</height>
    </rect>
   </property>
   <property name="title">
    <string>滤波选择</string>
   </property>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>30</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>滤波器：</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_2">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>30</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string/>
     </property>
    </item>
    <item>
     <property name="text">
      <string>BoxFilter</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>MeanBlur</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>GaussianBlur</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>MedianBlur</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>BilateralFilter</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QGroupBox" name="groupBox_4">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>500</y>
     <width>341</width>
     <height>91</height>
    </rect>
   </property>
   <property name="title">
    <string>显示设置</string>
   </property>
   <widget class="QCheckBox" name="checkBox">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>40</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>显示输出图</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>60</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>690</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>执行</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_2">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>690</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>取消</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_3">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>690</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>确认</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_15">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1090</width>
     <height>10</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(85, 170, 255);</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_16">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>10</y>
     <width>10</width>
     <height>730</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(85, 170, 255);</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_17">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>730</y>
     <width>1080</width>
     <height>10</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(85, 170, 255);</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_18">
   <property name="geometry">
    <rect>
     <x>1080</x>
     <y>10</y>
     <width>10</width>
     <height>730</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(85, 170, 255);</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_19">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>10</y>
     <width>10</width>
     <height>720</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(85, 170, 255);</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_20">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>270</y>
     <width>54</width>
     <height>12</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>10</y>
     <width>700</width>
     <height>720</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_21">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>110</y>
     <width>360</width>
     <height>10</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(85, 170, 255);</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_22">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>470</y>
     <width>360</width>
     <height>10</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(85, 170, 255);</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Filter_Function</receiver>
   <slot>on_run_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>569</y>
    </hint>
    <hint type="destinationlabel">
     <x>36</x>
     <y>536</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>Filter_Function</receiver>
   <slot>on_sure_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>241</x>
     <y>573</y>
    </hint>
    <hint type="destinationlabel">
     <x>170</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Filter_Function</receiver>
   <slot>on_cancel_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>323</x>
     <y>566</y>
    </hint>
    <hint type="destinationlabel">
     <x>329</x>
     <y>535</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>activated(QString)</signal>
   <receiver>Filter_Function</receiver>
   <slot>activate_comboBox_triggred()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>293</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>349</x>
     <y>8</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_2</sender>
   <signal>activated(QString)</signal>
   <receiver>Filter_Function</receiver>
   <slot>activate_comboBox_2_triggred()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>304</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>355</x>
     <y>163</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BoxFilterlSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Filter_Function</receiver>
   <slot>set_boxfilter_value(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>336</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>356</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MeanBlurSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Filter_Function</receiver>
   <slot>set_meanblur_value(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>GaussianBlurSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Filter_Function</receiver>
   <slot>set_gaussianblur_value(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>312</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>357</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MedianBlurSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Filter_Function</receiver>
   <slot>set_medianblur_value(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>369</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BilateralFilterSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Filter_Function</receiver>
   <slot>set_bilateralfilter(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>328</x>
     <y>399</y>
    </hint>
    <hint type="destinationlabel">
     <x>348</x>
     <y>434</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>Filter_Function</receiver>
   <slot>on_statechanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>155</x>
     <y>491</y>
    </hint>
    <hint type="destinationlabel">
     <x>14</x>
     <y>432</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_sure_clicked()</slot>
  <slot>on_run_clicked()</slot>
  <slot>on_cancel_clicked()</slot>
  <slot>activate_comboBox_triggred()</slot>
  <slot>activate_comboBox_2_triggred()</slot>
  <slot>set_boxfilter_value(int)</slot>
  <slot>set_meanblur_value(int)</slot>
  <slot>set_gaussianblur_value(int)</slot>
  <slot>set_medianblur_value(int)</slot>
  <slot>set_bilateralfilter(int)</slot>
  <slot>on_statechanged(int)</slot>
 </slots>
</ui>
